### PR TITLE
[CSS21][CSS22] Add regression test for position:fixed edge case

### PR DESCRIPTION
**Pull request by cvrebert**
Refs http://wkbug.com/147284, https://github.com/twbs/bootstrap/issues/16814

[Review on Reviewable](https://reviewable.io/reviews/w3c/csswg-test/813)

### CODE REVIEW

**Comment by cvrebert** [comment]
Adding feedback from https://lists.w3.org/Archives/Public/public-css-testsuite/2015Aug/0003.html

**Comment by cvrebert** [comment]
> There is already a test with the filename position-fixed-001
> 
> http://test.csswg.org/suites/css2.1/nightly-unstable/html4/position-fixed-001.htm
> 
> in the test suite. Fortunately for you, your test, in my opinion, is really about 'left: auto' offset. So, in all fairness, you were destined for a test filename change anyway.

File: css21/visuren/position-fixed-001-ref.xht [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html SYSTEM "about:legacy-compat">
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001-ref.xht]: > Please use
> ```
> <!DOCTYPE html>
> <meta charset="utf-8">
> ```

File: css21/visuren/position-fixed-001.xht [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html SYSTEM "about:legacy-compat">
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > Please use
> ```
> <!DOCTYPE html>
> <meta charset="utf-8">
> ```

File: css21/visuren/position-fixed-001.xht [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>CSS Test: position:fixed box's position calculated using absolute model</title>
+    <link rel="author" title="Chris Rebert" href="http://chrisrebert.com" />
+    <link rel="help" href="https://drafts.csswg.org/css2/visuren.html#choose-position" />
+    <link rel="help" href="https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-width" />
+    <link rel="help" href="https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-height" />
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > In my opinion, the last one (visudet.html#abs-non-replaced-height) is not relevant to your test.

File: css21/visuren/position-fixed-001.xht [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html SYSTEM "about:legacy-compat">
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>CSS Test: position:fixed box's position calculated using absolute model</title>
+    <link rel="author" title="Chris Rebert" href="http://chrisrebert.com" />
+    <link rel="help" href="https://drafts.csswg.org/css2/visuren.html#choose-position" />
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > Do not link to draft spec as they can change or will change in the 
future.

File: css21/visuren/position-fixed-001.xht [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    <link rel="help" href="https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-width" />
+    <link rel="help" href="https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-height" />
+    <link rel="match" href="position-fixed-001-ref.xht" />
+    <meta name="flags" content="" />
+    <meta name="assert" content="An element which is position:fixed should have its position calculated correctly using the absolute model when its top/bottom/left/right properties all use the default `auto` value and its parent is position:relative." />
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > Avoid use of expressions like "should work correctly", "should be 
implemented correctly", "should be supported correctly", "should be 
calculated correctly" and expressions like those in the meta text assert 
because they do not describe or explain in useful terms what the test is 
supposed to be testing, checking.

File: css21/visuren/position-fixed-001.xht [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    <style type="text/css">
+#shifted-column {
+    left: 50%;
+    width: 50%;
+    position: relative;
+    float: left;
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > Please explain what the float declaration is supposed to be doing in that test. Is it really required by your test? Does it do something in particular in your test? Is it a necessary and relevant component of your test?
> I believe the 'float: left' declaration should not be in your test at 
all.

File: css21/visuren/position-fixed-001.xht [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+}
+#red {
+  background-color: red;
+  position: fixed;
+}
+#green {
+  background-color: green;
+  position: absolute;
+  left: 50%;
+  z-index: 2;
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > you do not need to set a z-index since "Boxes with the same stack level in a stacking context are stacked back-to-front according to document tree order."

File: css21/visuren/position-fixed-001.xht [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+}
+p {
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > It is recommended to start all tests with the pass-fail-conditions sentence and, since it should not be part of the test itself, then we do not want to style it in any way. Sometimes, this is not possible but, I'd say 99% of the time, it is possible.

File: css21/visuren/position-fixed-001.xht [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    left: 50%;
+    width: 50%;
+    position: relative;
+    float: left;
+}
+.square {
+  width: 60px;
+  height: 60px;
+}
+#red {
```
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: > 'left: auto' declaration was added on purpose, deliberately, even 
though we know it is the default, initial box offset.

**Comment by cvrebert** [comment]
> One last thing. One other equivalent test could be created with 'direction: rtl', 'position fixed', 'right: auto' for many reasons. Also, same 2 tests but with 'position: absolute'. So, 3 other additional tests with the same basis code are possible here.

File: css21/visuren/position-fixed-001.xht (hunk 9 above)
Comment (cvrebert) [inline css21/visuren/position-fixed-001.xht]: Done.

**Comment by cvrebert** [comment]
>> I'm not familiar enough with `direction`/RTL to write those variants,
>
> http://www.gtalbot.org/BrowserBugsSection/css21testsuite/right-offset-position-fixed-001.xht
>
> iPhone 6, Safari 7+ fail this additional test.
> The text assert and a few other things in that right-offset-position-fixed-001 would need to be adjusted, tweaked.

**Comment by cvrebert** [comment]
>>> It is recommended to start all tests with the pass-fail-conditions sentence and, since it should not be part of the test itself, then we do not want to style it in >any way. Sometimes, this is not possible but, I'd say 99% of the time, it is possible.
>>
>> The position:fixed makes doing this somewhat tricky.
>> If you have a specific suggestion for how to achieve this, I'm fine
>> with changing the test, but I myself don't have any ideas.
>
> The modified test I created was achieving this.
>
> http://www.gtalbot.org/BrowserBugsSection/css21testsuite/left-offset-position-fixed-001.xht

**Comment by cvrebert** [comment]
Addressed all comments. Now for another round of review.

**Comment by cvrebert** [comment]
From https://lists.w3.org/Archives/Public/public-css-testsuite/2015Sep/0014.html :
> Chris, for many reasons, I propose you just submit your position-fixed tests (along with the reference files) and leave out, drop the abs-pos tests. I approve your
> left-offset-position-fixed-001.xht
> and
> right-offset-position-fixed-001.xht
> along with their respective reference files. Please add the date like this:
> ```
>    <link rel="reviewer" title="Gérard Talbot" href="http://www.gtalbot.org/BrowserBugsSection/css21testsuite/" /> 
> ```

**Comment by cvrebert** [comment]
Added the requested datestamps and removed the `position:absolute` variants.

**Comment by cvrebert** [comment]
@tabatkins Sorry to bother you, but could I persuade you to merge this PR, since Gérard doesn't use Git(Hub) and isn't familiar with importing patches via Mercurial? He's already reviewed & approved the 2 tests; see ML post linked above. Many thanks.

**Comment by cvrebert** [comment]
@Ms2ger Thank you!
